auth.js

```diff
@@ -0,0 +1,154 @@
+    emailInput.classList.add("error-box");
+    emailWrongError.classList.remove("show");
+  } else if (!emailInput.checkValidity()) {
+    emailNoneError.classList.remove("show");
+    emailInput.classList.add("error-box");
+    emailWrongError.classList.add("show");
+  } else {
+    emailNoneError.classList.remove("show");
+    emailWrongError.classList.remove("show");
+    emailInput.classList.remove("error-box");
+  }
+};
+emailInput.addEventListener("focusout", handleEmailFocusout);
+
+// input 비밀번호 error 처리
+const passwordNoneError = document.querySelector(
+  ".error-message.password-none"
+);
+const passwordWrongError = document.querySelector(
+  ".error-message.password-wrong"
+);
+
+let passwordValue = "";
+
+const handlePasswordFocusout = (e) => {
+  passwordValue = e.target.value.trim();
+  if (passwordValue.length === 0) {
+    passwordNoneError.classList.add("show");
+    passwordInput.classList.add("error-box");
+    passwordWrongError.classList.remove("show");
+  } else if (!passwordInput.checkValidity()) {
+    passwordNoneError.classList.remove("show");
+    passwordInput.classList.add("error-box");
+    passwordWrongError.classList.add("show");
+  } else {
+    passwordNoneError.classList.remove("show");
+    passwordWrongError.classList.remove("show");
+    passwordInput.classList.remove("error-box");
+  }
+};
+passwordInput.addEventListener("focusout", handlePasswordFocusout);
+
+// input 비밀번호 view 처리
+const passwordNoViewIcon = document.querySelector(".no-view-bt-icon");
+const passwordViewIcon = document.querySelector(".view-bt-icon");
+passwordNoViewIcon.classList.add("show");
+
+const handlePasswordView = () => {
+  const isHidden = passwordInput.type === "password";
+
+  if (isHidden) {
+    passwordInput.type = "text";
+    passwordViewIcon.classList.add("show");
+    passwordNoViewIcon.classList.remove("show");
+  } else {
+    passwordInput.type = "password";
+    passwordNoViewIcon.classList.add("show");
+    passwordViewIcon.classList.remove("show");
+  }
+};
+passwordNoViewIcon.addEventListener("click", handlePasswordView);
+passwordViewIcon.addEventListener("click", handlePasswordView);
+
+// input 비밀번호 확인 view 처리
+const passwordConfirmNoViewIcon = document.querySelector(
+  ".no-view-bt-icon.confirm"
+);
+const passwordConfirmViewIcon = document.querySelector(".view-bt-icon.confirm");
+passwordConfirmNoViewIcon.classList.add("show");
+
+const handlePasswordConfirmView = () => {
+  const isHidden = passwordConfirmInput.type === "password";
+
+  if (isHidden) {
+    passwordConfirmInput.type = "text";
+    passwordConfirmViewIcon.classList.add("show");
+    passwordConfirmNoViewIcon.classList.remove("show");
+  } else {
+    passwordConfirmInput.type = "password";
+    passwordConfirmNoViewIcon.classList.add("show");
+    passwordConfirmViewIcon.classList.remove("show");
+  }
+};
+if (passwordConfirmNoViewIcon) {
+  passwordConfirmNoViewIcon.addEventListener(
+    "click",
+    handlePasswordConfirmView
+  );
+}
```

Comment:
💬 여담
변수가 null 또는 undefined일 때만 안전하게 넘어가도록 하려면 optional chaining을 활용하는 것도 좋은 방법입니다.
이렇게 작성하면 존재 여부를 따로 조건문으로 확인하지 않으므로 코드가 조금 더 간결해집니다.

[mdn](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

```suggestion
passwordConfirmNoViewIcon?.addEventListener(
  "click",
  handlePasswordConfirmView
);
```